src/useStorage/storage.ts

```diff
@@ -34,7 +34,11 @@ export class StorageManager {
       return defaultValue
     }
 
-    return JSON.parse(value) as T
+    try {
+      return JSON.parse(value) as T
+    } catch {
+      return defaultValue
+    }
```

Comment:
@stackoverfloweth swallowing this error seems like a bad idea. 

I'm also wondering if we shouldn't return `value ?? defaultValue`? That way we shouldn't need to do anything special to support migrating old storage values.

Reply:
@pleek91 I don't follow, can you show me an example of what you're thinking?